tests/multi_ptr/multi_ptr_access_members.h

```diff
@@ -0,0 +1,156 @@
+ * @param address_space_name Current sycl::access::address_space string
+ *        representation
+ * @param is_decorated_name Current sycl::access::decorated string
+ *        representation
+ */
+template <typename T, typename AddrSpaceT, typename IsDecorated>
+class run_access_members_tests {
+  static constexpr sycl::access::address_space space = AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecorated::value;
+  using multi_ptr_t = const sycl::multi_ptr<T, space, decorated>;
+
+ public:
+  void operator()(const std::string &type_name,
+                  const std::string &address_space_name,
+                  const std::string &is_decorated_name) {
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    constexpr int val_to_init = 42;
+    T value = user_def_types::get_init_value_helper<T>(val_to_init);
+    bool dereference_result = false;
+    bool dereference_op_result = false;
+    bool get_result = false;
+    bool get_raw_result = false;
+    bool get_decorated_result = false;
+    {
+      sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+      sycl::buffer<bool, 1> dereference_result_buf(&dereference_result,
+                                                   sycl::range<1>(1));
+      sycl::buffer<bool, 1> dereference_op_result_buf(&dereference_op_result,
+                                                      sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_result_buf(&dereference_op_result,
+                                           sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_raw_result_buf(&dereference_op_result,
+                                               sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_decorated_result_buf(&dereference_op_result,
+                                                     sycl::range<1>(1));
+
+      queue.submit([&](sycl::handler &cgh) {
+        auto dereference_result_acc =
+            dereference_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto dereference_op_result_acc =
+            dereference_op_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto get_result_acc =
+            get_result_buf.template get_access<sycl::access_mode::write>(cgh);
+        auto get_raw_result_acc =
+            get_raw_result_buf.template get_access<sycl::access_mode::write>(
+                cgh);
+        auto get_decorated_result_acc =
+            get_decorated_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+
+        auto acc_for_multi_ptr =
+            val_buffer.template get_access<sycl::access_mode::read>(cgh);
+        cgh.single_task([=] {
+          const multi_ptr_t multi_ptr(acc_for_multi_ptr);
+
+          // Check dereference operator return type correctness
+          dereference_result_acc[0] =
+              std::is_same_v<decltype(*multi_ptr), multi_ptr::reference>;
```

Comment:
I have felling that you have to specify template parameters to get access to class member types.
godbolt example: https://godbolt.org/z/zaqq4baKT
```suggestion
              std::is_same_v<decltype(*multi_ptr), multi_ptr<T, AddrSpaceT, IsDecorated>::reference>;
```

Reply:
yes, you are right, type should be here
I've done this [here](https://github.com/KhronosGroup/SYCL-CTS/pull/346/commits/4bcaf862cf82c80d669077c2eb2ffc920474524a#diff-7c3b5239659a41d2cc6341067d6094f03525239f4a9380fce5a1056b6bfddc62L92-R133)